drive-stack/ota-agent/src/main.rs

```diff
@@ -0,0 +1,459 @@
+use std::{
+    collections::HashMap,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    path::{Path, PathBuf},
+    sync::Arc,
+    thread,
+    time::Duration,
+};
+
+use anyhow::{bail, Context, Result};
+use argh::FromArgs;
+use bytes::Buf; // for chunk.chunk()
+use chrono::Utc;
+use futures_util::{StreamExt, TryStreamExt};
+use hex;
+use hostname::get as get_hostname;
+use if_addrs::get_if_addrs;
+use net_detec::Server as MdnsServer;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use thiserror::Error;
+use tokio::{fs, io::AsyncWriteExt};
+use tracing::{info, error};
+use tracing_subscriber::EnvFilter;
+use warp::{http::StatusCode, Filter};
+
+use conUDS::modules::uds::{FlashStatus, UdsSession};
+
+/// Host a Rest API with ability to upload and deploy applications
+#[derive(FromArgs, Debug)]
+struct Args {
+    /// the CAN device to use. `can0` is used if this option is not provided
+    #[argh(option, short = 't', default = "String::from(\"can0\")")]
+    device: String,
+
+    /// the manifest of UDS can nodes on the bus
+    #[argh(option, short = 'm', default = "String::from(\"drive-stack/conUDS/nodes.yml\")")]
+    node_manifest: String,
+
+    /// network interface to use (e.g., eth0, en0)
+    #[argh(option, short = 'i')]
+    interface: Option<String>,
+
+    /// IPv4 to advertise. If omitted, the server will detect the IPv4 of the chosen interface.
+    #[argh(option)]
+    ip: Option<String>,
+
+    /// service type like _carputer-uds._tcp.local.
+    #[argh(option, short = 's', default = "String::from(\"_ota-agent._tcp.local.\")")]
+    service_name: String,
+
+    /// port to advertise
+    #[argh(option, default = "10000")]
+    port: u16,
+
+    /// directory where received binaries will be saved (and overwritten) before flashing
+    #[argh(option)]
+    save_dir: PathBuf,
+}
```

Comment:
Do we _have_ to do this with args? It would be great to just hardcode everything honestly, less room for someone to make a mistake somewhere